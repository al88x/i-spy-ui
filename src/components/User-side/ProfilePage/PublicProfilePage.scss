@import '../../general/general-styling.scss';

.profilePage {

    .reportSightingButton {
        text-align: center;

        .indivButton {
            @include indivButton;
            width:80%;
            font-size: 20px;
            padding-left: 50px;
            padding-right: 50px;

            @media screen and (min-width: $tablet-breakpoint) {
                width:70%;
                font-size:30px;
                padding-left: 100px;
                padding-right: 100px;
            }

            @media screen and (min-width: $desktop-breakpoint) {
                width:60%;
                font-size:35px;
                padding-left: 200px;
                padding-right: 200px;
            }
                
        }
    }
}