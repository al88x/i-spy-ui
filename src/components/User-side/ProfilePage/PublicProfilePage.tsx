import React from 'react';
import { Link } from "react-router-dom";
import './PublicProfilePage.scss';

export function PublicProfilePage() {
    return (
        <div className="profilePage">
        <div className="reportSightingButton">
            <Link className="indivButton" to="/profile/createReport">REPORT A SIGHTING</Link>
        </div>
        </div>
    )
}