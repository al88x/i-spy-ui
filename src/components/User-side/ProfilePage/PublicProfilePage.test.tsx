import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route } from "react-router-dom";
import { PublicProfilePage } from './PublicProfilePage';

import { render, wait } from '@testing-library/react';

describe("ProfilePage", () => {
    it("Should render a Reporting A Sighting button that matches snapshot", () => {
        const component = renderer.create(
            <MemoryRouter initialEntries={["/profile/1"]}>
                <PublicProfilePage/>
            </MemoryRouter>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });
});

describe('testing api', () => {

    it("should render Fetching data... while waiting", () => {
        const profilePage = render(
            <MemoryRouter initialEntries={["/profile/1"]}>
                <Route path="/profile/:id" component={PublicProfilePage}/>
            </MemoryRouter>
        );
        expect(profilePage.getByText("Fetching data...")).toBeInTheDocument();
    })

    it("should render the response when returns correctly", async () => {
        const profilePage = render(
            <MemoryRouter initialEntries={["/profile/1"]}>
                <Route path="/profile/:id" component={PublicProfilePage}/>
            </MemoryRouter>
        );
        await wait(() => expect(profilePage.getByText("Male")).toBeInTheDocument);
    })

    // it("should shows an error message if the api call fails", async () => {
    //     mockFailedFetch();
    //     const history = createBrowserHistory();

    //     const profilePage = render(<Router history={history}><PublicProfilePage/></Router>);
    //     await wait(() => expect(profilePage.getByText("Oh No!!! There was an error")).toBeInTheDocument());
    // });

});
