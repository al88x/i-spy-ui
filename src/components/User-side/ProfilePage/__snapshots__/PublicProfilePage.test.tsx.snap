// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage Should render a Reporting A Sighting button that matches snapshot 1`] = `
<div>
  Fetching data...
</div>
`;
